### PR TITLE
Support blocking lid space compaction during high remove rate

### PR DESCRIPTION
**Pull request by geirst**
@toregge and @baldersheim please review

During a period with a high rate of remove operations, there is no use running lid space compaction
as this will interfere with the remove operations, increasing latency of those.
Moving a document as part of lid space compaction is a costly operation (similar to putting the document in the first place) and it typically uses both the index and attribute writer thread pools.

### CODE REVIEW

File: searchcore/src/vespa/searchcore/proton/documentmetastore/documentmetastore.h [hunk 1]
```diff
@@ -71,7 +74,7 @@ class DocumentMetaStore final : public DocumentMetaStoreAttribute,
     uint32_t            _shrinkLidSpaceBlockers;
     const SubDbType     _subDbType;
     bool                _trackDocumentSizes;
-    search::LidUsageStats::TimePoint _last_remove_batch;
+    documentmetastore::OperationListener::SP _op_listener;
```
Comment (toregge) [inline searchcore/src/vespa/searchcore/proton/documentmetastore/documentmetastore.h:77]: Consider using std::shared_ptr&lt;documentmetastore::OperationListener&gt;, to be consistent with forward declaration above
and not depend on corresponding include file having been included.

**Review by toregge — APPROVED** [review]

File: searchcore/src/vespa/searchcore/proton/documentmetastore/i_document_meta_store.h [hunk 2]
```diff
@@ -82,6 +83,8 @@ struct IDocumentMetaStore : public search::IDocumentMetaStore,
      */
     virtual void compactLidSpace(DocId wantedLidLimit) = 0;
 
+    virtual void set_operation_listener(documentmetastore::OperationListener::SP op_listener) = 0;
```
Comment (toregge) [inline searchcore/src/vespa/searchcore/proton/documentmetastore/i_document_meta_store.h:86]: Consider forward declaring documentmetastore::OperationListener and using std::shared_ptr here.

File: searchcore/src/vespa/searchcore/proton/documentmetastore/i_document_meta_store.h [hunk 3]
```diff
@@ -4,6 +4,7 @@
 
 #include "lid_gid_key_comparator.h"
 #include "i_simple_document_meta_store.h"
+#include "operation_listener.h"
```
Comment (toregge) [inline searchcore/src/vespa/searchcore/proton/documentmetastore/i_document_meta_store.h:7]: This include is not needed if documentmetastore::OperationListener is forwared declared later on in this file.

File: searchcore/src/vespa/searchcore/proton/documentmetastore/documentmetastore.h (hunk 1 above)
Comment (geirst) [reply searchcore/src/vespa/searchcore/proton/documentmetastore/documentmetastore.h]: Fixed

File: searchcore/src/vespa/searchcore/proton/documentmetastore/i_document_meta_store.h (hunk 2 above)
Comment (geirst) [reply searchcore/src/vespa/searchcore/proton/documentmetastore/i_document_meta_store.h]: Fixed

File: searchcore/src/vespa/searchcore/proton/documentmetastore/i_document_meta_store.h (hunk 3 above)
Comment (geirst) [reply searchcore/src/vespa/searchcore/proton/documentmetastore/i_document_meta_store.h]: Fixed

**Review by toregge — APPROVED** [review]

**Review by baldersheim — APPROVED** [review]

**Pull request merged**